### PR TITLE
Avoid global rngs in testsuite

### LINKED ISSUE #305: Avoid dependence on global NumPy rng in testsuite
The testsuite provides its own rng (https://github.com/Happy-Algorithms-League/hal-cgp/blob/master/test/conftest.py#L13). To avoid difficult to reproduce testsuite errors, we should always prefer this rng over the global one (`np.random`).

### PR DESCRIPTION
**Pull request by jakobj**
- replace numpy global rng
- replace torch global rng

closes #305

### CODE REVIEW

File: test/test_hl_api.py [hunk 1]
```diff
@@ -12,16 +12,15 @@ def _objective_test_population(individual, rng_seed):
...
-
     n_function_evaluations = 100
 
     f_graph = individual.to_func()
 
     def f_target(x):  # target function
         return x[:, 0] - x[:, 1]
 
-    x = np.random.normal(size=(n_function_evaluations, 2))
+    rng = np.random.RandomState(seed=rng_seed)
```
Comment (mschmidt87) [inline test/test_hl_api.py:22]: Why don't you use the fixture here?

File: test/test_hl_api.py [hunk 2]
```diff
@@ -81,6 +78,8 @@ def _objective(individual):
         if not individual.fitness_is_None():
             return individual
 
+        rng = np.random.RandomState(rng_seed)
```
Comment (mschmidt87) [inline test/test_hl_api.py:81]: same here: why not the fixture?

**Review by mschmidt87 — CHANGES_REQUESTED** [review]
Good work. But why don't you consistently use the `rng` fixture everywhere?

File: test/test_hl_api.py (hunk 1 above)
Comment (jakobj) [reply test/test_hl_api.py:22]: I think the intention here is to use the same random number stream in every call to the objective such that the fitness does *not* depend on the random number stream. I think if i'd use the fixture, then rngs are shared across objective calls, right? same reason below

**Comment by jakobj** [comment]
ping @mschmidt87

**Review by mschmidt87 — APPROVED** [review]
LGTM. I think zou need to push once again to trigger the new testsuite including Python 3.9

**Pull request merged**